fix(react-grid): correct DataTypeProvider.ValueEditorProps types (#3146)
DevExpress/devextreme-reactive: packages/dx-react-grid/docs/reference/data-type-provider.md

value | any | Specifies the editor value.
onValueChange | (newValue: any) => void | Handles value changes.

> disabled | boolean | **true** if users should not be able to edit the value, **false** otherwise.

Reviewer: In the majority of our documentation, we use "`otherwise **false**`" instead of "`**false** otherwise`". Can we update this as follows (and in the descriptions below as well)?

```suggestion
disabled | boolean | **true** if users should not be able to edit the value; otherwise **false**.
```

> autoFocus | boolean | **true** if editor should be focused on render, **false** otherwise.

Reviewer: ```suggestion
autoFocus | boolean | **true** if the editor should be focused on render; otherwise, **false**.
```

> onBlur | () => void | Switches editor from editing state on editor lose focus (uses for Inline Cell Editing).

Reviewer 2: ```suggestion
onBlur | () => void | Switches the editor from edit state when it loses focus (used for Inline Cell Editing).
```

> onFocus | () => void | Apply action on editor get focus (uses for Inline Cell Editing).

Reviewer 2: ```suggestion
onFocus | () => void | Applies an action when the editor gets focus (used for Inline Cell Editing).
```

> onKeyDown | () => void | Handles key press: Enter to save changes, or Esc to discard the changes and deactivate the cell editor (uses for Inline Cell Editing).

Reviewer 2: ```suggestion
onKeyDown | () => void | Handles key press: Enter saves changes, Esc discards them and switches the editor from edit state (used for Inline Cell Editing).
```
